pkg/platform/kube/platform.go

```diff
@@ -1414,27 +1410,28 @@ func (p *Platform) generateFunctionToAPIGatewaysMapping(ctx context.Context, nam
 	return functionToAPIGateways, nil
 }
 
-func (p *Platform) getApiGatewaysForFunction(ctx context.Context,
-	namespace string,
-	functionName string) ([]nuclioio.NuclioAPIGateway, error) {
-	var functionsApiGateways []nuclioio.NuclioAPIGateway
+func (p *Platform) getApiGateways(ctx context.Context, getAPIGatewaysOptions *platform.GetAPIGatewaysOptions) ([]nuclioio.NuclioAPIGateway, error) {
 
 	apiGateways, err := p.consumer.NuclioClientSet.NuclioV1beta1().
-		NuclioAPIGateways(namespace).
+		NuclioAPIGateways(getAPIGatewaysOptions.Namespace).
 		List(ctx, metav1.ListOptions{})
 	if err != nil {
 		return nil, errors.Wrap(err, "Failed to list API gateways")
 	}
-	for _, apiGateway := range apiGateways.Items {
-		for _, upstream := range apiGateway.Spec.Upstreams {
+	if getAPIGatewaysOptions.FunctionName != "" {
+		var functionsApiGateways []nuclioio.NuclioAPIGateway
+		for _, apiGateway := range apiGateways.Items {
+			for _, upstream := range apiGateway.Spec.Upstreams {
 
-			if upstream.NuclioFunction.Name == functionName {
-				functionsApiGateways = append(functionsApiGateways, apiGateway)
-				break
+				if upstream.NuclioFunction.Name == getAPIGatewaysOptions.FunctionName {
```

Comment:
not sure, but wdyt about also filtering allowing with partial names?
with regex style `func-*`?

Reply:
@TomerShor tbh, currently I can't think about a scenario where it can be that useful, so not sure if it's neccesary